src/cfnlint/data/ExtendedSpecs/all/03_value_types.json

```diff
@@ -0,0 +1,332 @@
+[
+  {
+    "op": "add",
+    "path": "/ValueTypes",
+    "value": {
+      "AllocationId": {
+        "GetAtt": {
+          "AWS::EC2::EIP": "AllocationId"
+        },
+        "Ref": {
+          "Parameters": [
+            "String"
+          ]
+        }
+      },
+      "AvailabilityZone": {
+        "GetAtt": {
+          "AWS::EC2::Instance": "AvailabilityZone",
+          "AWS::EC2::Subnet": "AvailabilityZone",
+          "AWS::OpsWorks::Instance": "AvailabilityZone"
+        },
+        "Ref": {
+          "Parameters": [
+            "String",
+            "AvailabilityZone"
+          ]
+        }
+      },
+      "AvailabilityZones": {
+        "Ref": {
+          "Parameters": [
+            "Strings",
+            "AvailabilityZones"
+          ]
+        }
+      },
+      "EbsVolumeType": {
+        "AllowedValues": [
+          "gp2",
+          "io1",
+          "sc1",
+          "st1",
+          "standard"
+        ]
+      },
+      "IamInstanceProfile": {
+        "Ref": {
+          "Parameters": [
+            "String"
+          ],
+          "Resources": [
+            "AWS::IAM::InstanceProfile"
+          ]
+        }
+      },
+      "IamInstanceProfile.Arn": {
+        "GetAtt": {
+          "AWS::IAM::InstanceProfile": "Arn"
+        },
+        "Ref": {
+          "Parameters": [
+            "String"
+          ]
+        }
+      },
+      "IamRole": {
+        "Ref": {
+          "Parameters": [
+            "String"
+          ],
+          "Resources": [
+            "AWS::IAM::Role"
+          ]
+        }
+      },
+      "IamRole.Arn": {
+        "GetAtt": {
+          "AWS::IAM::Role": "Arn"
+        },
+        "Ref": {
+          "Parameters": [
+            "String"
+          ]
+        }
+      },
+      "IamUser.Arn": {
+        "GetAtt": {
+          "AWS::IAM::User": "Arn"
+        },
+        "Ref": {
+          "Parameters": [
+            "String"
+          ]
+        }
+      },
+      "ImageId": {
+        "Ref": {
+          "Parameters": [
+            "String",
+            "ImageId"
+          ]
+        }
+      },
+      "KeyPair": {
+        "Ref": {
+          "Parameters": [
+            "String",
+            "KeyPair"
+          ]
+        }
+      },
+      "KmsKey": {
+        "GetAtt": {
+          "AWS::KMS::Key": "Arn"
+        },
+        "Ref": {
+          "Parameters": [
+            "String"
+          ]
+        },
+        "Resources": [
+          "AWS::KMS::Key"
+        ]
+      },
+      "LambdaRuntime": {
+        "AllowedValues": [
+          "dotnetcore1.0",
+          "dotnetcore2.0",
+          "dotnetcore2.1",
+          "go1.x",
+          "java8",
+          "nodejs",
+          "nodejs4.3-edge",
+          "nodejs4.3",
+          "nodejs6.10",
+          "nodejs8.10",
+          "provided",
+          "python2.7",
+          "python3.6",
+          "python3.7",
+          "ruby2.5"
```

Comment:
You are still missing `provided`.  https://docs.aws.amazon.com/lambda/latest/dg/API_CreateFunction.html#SSS-CreateFunction-request-Runtime